docs/reference/builder.md

```diff
@@ -530,8 +530,10 @@ FROM extras:${CODE_VERSION}
 CMD  /code/run-extras
 ```
 
-To use the default value of an `ARG` declared before the first `FROM` use an
-`ARG` instruction without a value:
+An `ARG` declared before a `FROM` is outside of a build stage, so it
```

Comment:
Maybe something like "An `ARG` instructions is only relevant for the next `FROM` instruction that comes after it in the Dockerfile, and then using this example to illustrate that.